python/spec/dependabot/python/update_checker/pip_compile_version_resolver_spec.rb

```diff
@@ -301,7 +301,7 @@
         it "raises a helpful error" do
           expect { subject }.
             to raise_error(Dependabot::GitDependencyReferenceNotFound) do |err|
-              expect(err.dependency).to eq("pythonfinder")
+              expect(err.dependency).to eq("v15.1.2")
```

Comment:
This looks to be the same as [this test](https://github.com/dependabot/dependabot-core/blob/15e9b79554da14a9f9ecc0f79e6cbe3f6a022fa2/python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb#L299). Do you know why that would pass as-is, but this needed a change?

Reply:
Seems to be thrown here when a hash can't be extracted for the branch: https://github.com/dependabot/dependabot-core/blob/8c448448c8b1ada0e9de2c87f1fba043eca7f5d1/common/lib/dependabot/git_commit_checker.rb#L86

The linked test is called "for another dependency, that has a bad ref" so that seems to add up. 

It's a different code path but uses the same exception, but in that case the dependency has already been identified. Here it was being extracted from some output.

Reply:
I think this will result in some confusing error messages, we display this `error.dependency` as:

```
Dependabot couldn't fetch the branch/reference for #{error.dependency}

"Your dependency file specified a branch or reference for "\
          "`#{error.dependency}`, but Dependabot couldn't find it at the "\
          "project's source. Has it been removed?"
```

If there is another way to determine the dependency causing the issue, it might be worth pulling on that thread a bit